tests/api2/pool_dataset_encryption.py

```diff
@@ -0,0 +1,870 @@
+disk_pool = sorted(list(set(disk_list) - set(nas_disk)))
+# genrated token_hex 32bit for
+pool_token_hex = secrets.token_hex(32)
+pool_token_hex2 = secrets.token_hex(32)
+dataset_token_hex = secrets.token_hex(32)
+dataset_token_hex2 = secrets.token_hex(32)
+dataset = f'{pool_name}/encrypted'
+dataset_url = dataset.replace('/', '%2F')
+child_dataset = f'{dataset}/child'
+child_dataset_url = child_dataset.replace('/', '%2F')
+Reason = 'Skip for HA'
+skip_for_ha = pytest.mark.skipif(ha, reason=Reason)
+
+
+@skip_for_ha
+def test_01_create_a_normal_pool():
+    global pool_id
+    payload = {
+        'name': pool_name,
+        'encryption': False,
+        'topology': {
+            'data': [
+                {'type': 'STRIPE', 'disks': disk_pool}
+            ],
+        }
+    }
+    results = POST('/pool/', payload)
+    assert results.status_code == 200, results.text
+    job_id = results.json()
+    job_status = wait_on_job(job_id, 120)
+    assert job_status['state'] == 'SUCCESS', str(job_status['results'])
+    pool_id = job_status['results']['result']['id']
+
+
+def test_02_create_a_passphrase_encrypted_root_on_normal_pool():
+    payload = {
+        'name': dataset,
+        'encryption_options': {
+            'generate_key': False,
+            'pbkdf2iters': 100000,
+            'algorithm': 'AES-128-CCM',
+            'passphrase': 'my_passphrase',
+        },
+        'encryption': True,
+        'inherit_encryption': False
+    }
+    results = POST('/pool/dataset/', payload)
+    assert results.status_code == 200, results.text
+
+
+def test_03_add_the_comment_on_the_passphrase_encrypted_root():
+    payload = {
+        'comments': 'testing encrypted dataset'
+    }
+    results = PUT(f'/pool/dataset/id/{dataset_url}/', payload)
+    assert results.status_code == 200, results.text
+
+
+def test_04_change_a_passphrase_encrypted_root_key_encryption():
+    payload = {
+        'id': dataset,
+        'change_key_options': {
+            'key': dataset_token_hex,
+        }
+    }
+    results = POST(f'/pool/dataset/change_key/', payload)
+    assert results.status_code == 200, results.text
+    job_id = results.json()
+    job_status = wait_on_job(job_id, 120)
+    assert job_status['state'] == 'SUCCESS', str(job_status['results'])
+
+
+def test_05_delete_passphrase_encrypted_root():
+    results = DELETE(f'/pool/dataset/id/{dataset_url}/')
+    assert results.status_code == 200, results.text
+
+
+def test_06_create_not_encrypted_dataset_on_a_normal_pool():
+    payload = {
+        'name': dataset,
+        'encryption': False,
+    }
+    results = POST('/pool/dataset/', payload)
+    assert results.status_code == 200, results.text
+
+
+def test_07_delete_not_encrypted_dataset():
+    results = DELETE(f'/pool/dataset/id/{dataset_url}/')
+    assert results.status_code == 200, results.text
+
+
+def test_08_create_a_dataset_with_inherit_encryption_true_on_a_normal_pool():
```

Comment:
So the parent here is not encrypted, let's please verify that the dataset created states the same